### PR TITLE
Sign in result

### PR DESCRIPTION
**Pull request by guperrot**
Please have a look at our [guidelines for contributions](https://github.com/Microsoft/AppCenter-SDK-Android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Description

Replace void by SignInResult in signIn method.

### CODE REVIEW

File: sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/SignInResult.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+     * @param userInformation User information.
+     * @param exception       Exception if an error occurred during sign-in.
+     */
+    SignInResult(UserInformation userInformation, Exception exception) {
+        mUserInformation = userInformation;
+        mException = exception;
+    }
+
+    /**
+     * Get error if any occurred.
```
Comment (jaelim-ms) [inline sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/SignInResult.java]: We are returning an exception. *Get an exception of sign-in if it failed* maybe?

File: sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/SignInResult.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+     *
+     * @return Error that occurred during sign-in or null if user information is available.
```
Comment (jaelim-ms) [inline sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/SignInResult.java]: *Error* -> *Exception for an error*

File: apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/SettingsActivity.java [hunk 3]
```diff
@@ -521,6 +522,7 @@ public void onReset() {
             });
             initClickableSetting(R.string.clear_crash_user_confirmation_key, new Preference.OnPreferenceClickListener() {
 
+                @SuppressLint("VisibleForTests")
```
[the remark this answers is not in the document]
Comment (guperrot) [reply apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/SettingsActivity.java]: It's a constant that we didn't want to expose but somehow sasquatch always used it.
Now android studio has more warnings, its not related to my changes, was already there.

File: sdk/appcenter-identity/src/test/java/com/microsoft/appcenter/identity/IdentityTest.java [hunk 4]
```diff
@@ -461,6 +493,119 @@ public void readCacheAndRefreshNotModified() throws Exception {
...
+        verify(publicClientApplication).acquireToken(notNull(Activity.class), notNull(String[].class), callbackCaptor.capture());
+        callbackCaptor.getValue().onSuccess(mockAuthResult("idToken", "accountId", "homeAccountId"));
+
+        /* Verify disabled error. */
+        assertNotNull(future.get());
+        assertTrue(future.get().getException() instanceof IllegalStateException);
+        assertNull(future.get().getUserInformation());
+    }
+
+    private void mockReadyToSignIn() throws Exception {
```
[the remark this answers is not in the document]
Comment (guperrot) [reply sdk/appcenter-identity/src/test/java/com/microsoft/appcenter/identity/IdentityTest.java]: Only 1 of them can be static, updated

**Pull request merged**